### DISCUSSION TITLE
Stablizerのref_contact_state

### DISCUSSION THREAD

**Discussion by Naoki-Hiraoka** [YutaKojio/hrpsys-base#26] [Q&A] [answered]
Stablizerにref_contact_stateをセットするのはここですが、
https://github.com/YutaKojio/hrpsys-base/blob/5aab7f9809b1ef76dd62bf323e522834d84097be/rtc/AutoBalancer/AutoBalancer.cpp#L1179
それよりも前の箇所でref_contact_stateを用いてfootOriginCoordsを求める処理が行われるので、
https://github.com/YutaKojio/hrpsys-base/blob/5aab7f9809b1ef76dd62bf323e522834d84097be/rtc/AutoBalancer/AutoBalancer.cpp#L818
初回などにバグになる恐れがありそう?

**Comment by YutaKojio** [comment, marked as answer]
st->getActualParameters();
の中で使ってるのは，
https://github.com/YutaKojio/hrpsys-base/blob/f9a3710813a6c5d2fd9a5d645a055b8af52d2148/rtc/AutoBalancer/Stabilizer.cpp#L467
の部分だけかな？

ref_contact_states自体は
https://github.com/YutaKojio/hrpsys-base/blob/f9a3710813a6c5d2fd9a5d645a055b8af52d2148/rtc/AutoBalancer/Stabilizer.cpp#L198
で初期化されていて，
全部trueだから実態とはあってないかもしれないけど，
ここもactivateされた１制御周期目にstart-stしないから大丈夫かな
